### PR TITLE
Global gradient clipping

### PR DESCRIPTION
**Pull request by gobbleturk**

### CODE REVIEW

**Review by rwitten — CHANGES_REQUESTED** [review]
I think by requesting changes I can make this clearly with you having next steps.

File: MaxText/train.py [hunk 1]
```diff
@@ -187,9 +187,23 @@ def loss_fn(params):
...
   grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
-  (loss, intermediate_outputs), grads = grad_fn(state.params)
-  new_state = state.apply_gradients(grads=grads)
+  (loss, intermediate_outputs), raw_grads = grad_fn(state.params)
+  if config.gradient_clipping_threshold > 0:
+    grads, _ = optax.clip_by_global_norm(config.gradient_clipping_threshold).update(raw_grads, None, None)
+  else:
+    grads = raw_grads
+
+  opt = state.opt_state
```
[the remark this answers is not in the document]
Comment (gobbleturk) [reply MaxText/train.py]: Uh this is a good catch, we should maintain the same simple downstream API so I've reverted this to keep the same downstream `apply_gradients` API.

I was taking this from the int8 branch and convinced myself it was necessary, but after another look it's clearly not.

**Review by rwitten — APPROVED** [review]
LGTM
